### PR TITLE
tests: adding extra scenarios for fault injection

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change includes some scenarios that are missing to check fault injection.

In this case are included fault/reboot for:
- install snapd --dangerous
- remodel
- install kernel component
- update boot config

### CODE REVIEW

**Review by bboozzoo — APPROVED** [review]
LGTM

One more idea about possible scenarios is a transaction updating both the base snap and snapd. Depending on how far has the installation process advanced a failure should either roll back the complete thing, or finish updating both snaps.

File: tests/nested/core/core20-fault-inject-on-install/task.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    remote.exec "test -e /var/lib/snapd/faults/$TAG:$FAULT"
+
+    if [ "$FAULT" = panic ]; then
+        echo "Ensure the panic was injected and triggered stamp file is present"
+        remote.exec "sudo journalctl -u snapd | grep -A 3 panic" > output
+        MATCH "$TAG":$FAULT < output
+        MATCH "osutil.injectFault" < output
+    fi
+
+    FINAL_STATUS="$(remote.exec snap changes | grep -E "^$change_id .*" | awk ' {print $2} ')"
```
Comment (alfonsosanchezbeato) [inline tests/nested/core/core20-fault-inject-on-install/task.yaml:71]: But the final status is known depending on the fault, right? So we should now and check against the expected outcome. I know we are not very consistent and sometimes we fail and sometimes we don't, but I think that the test should test also changes in that behavior.

File: tests/nested/manual/core20-fault-inject-on-install-component/task.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    remote.exec "test -e /var/lib/snapd/faults/$TAG:$FAULT"
+
+    if [ "$FAULT" = panic ]; then
+        echo "Ensure the panic was injected and triggered stamp file is present"
+        remote.exec "sudo journalctl -u snapd | grep -A 3 panic" > output
+        MATCH "$TAG":$FAULT < output
+        MATCH "osutil.injectFault" < output
+    fi
+
+    FINAL_STATUS="$(remote.exec snap changes | grep -E "^$change_id .*" | awk ' {print $2} ')"
```
Comment (alfonsosanchezbeato) [inline tests/nested/manual/core20-fault-inject-on-install-component/task.yaml:84]: same regarding expected status

File: tests/nested/manual/core20-fault-inject-on-remodel/task.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    remote.exec "test -e /var/lib/snapd/faults/$TAG:$FAULT"
+
+    if [ "$FAULT" = panic ]; then
+        echo "Ensure the panic was injected and triggered stamp file is present"
+        remote.exec "sudo journalctl -u snapd | grep -A 3 panic" > output
+        MATCH "$TAG":$FAULT < output
+        MATCH "osutil.injectFault" < output
+    fi
+
+    FINAL_STATUS="$(remote.exec snap changes | grep -E "^$change_id .*" | awk ' {print $2} ')"
```
Comment (alfonsosanchezbeato) [inline tests/nested/manual/core20-fault-inject-on-remodel/task.yaml:105]: same

**Review by alfonsosanchezbeato — COMMENTED** [review]
I have a comment on what is the expected outcome of the tests

File: tests/nested/core/core20-fault-inject-on-install/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/nested/core/core20-fault-inject-on-install/task.yaml:72]: updated

File: tests/nested/manual/core20-fault-inject-on-install-component/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/nested/manual/core20-fault-inject-on-install-component/task.yaml:84]: updated

File: tests/nested/manual/core20-fault-inject-on-remodel/task.yaml (hunk 3 above)
Comment (sergiocazzolato) [reply tests/nested/manual/core20-fault-inject-on-remodel/task.yaml:105]: updaed

File: tests/nested/manual/core20-fault-inject-on-update-config/task.yaml [hunk 4]
```diff
@@ -9,8 +9,8 @@ systems: [ubuntu-2*]
 environment:
     TAG/gadget_panic_command_line: update-command-line-gadget
     FAULT/gadget_panic_command_line: panic
-    TAG/gadget_panic_config_bootloader: update-config-bootloader
-    FAULT/gadget_panic_config_bootloader: panic
+    #TAG/gadget_panic_config_bootloader: update-config-bootloader
+    #FAULT/gadget_panic_config_bootloader: panic
```
Comment (alfonsosanchezbeato) [inline tests/nested/manual/core20-fault-inject-on-update-config/task.yaml:13]: Is this the one that finishes with error? Maybe worth adding a comment on why it is commented out.

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, although I have a question

File: tests/nested/manual/core20-fault-inject-on-update-config/task.yaml (hunk 4 above)
Comment (sergiocazzolato) [reply tests/nested/manual/core20-fault-inject-on-update-config/task.yaml:13]: No, this is an scenario that cannot be reproduced, I'll work on that one in a following pr, I need @valentindavid for that

**Pull request merged**